unstructured_inference/models/yolox.py

```diff
@@ -2,208 +2,129 @@
 # Unstructured modified the original source code found at:
 # https://github.com/Megvii-BaseDetection/YOLOX/blob/237e943ac64aa32eb32f875faa93ebb18512d41d/yolox/data/data_augment.py
 # https://github.com/Megvii-BaseDetection/YOLOX/blob/ac379df3c97d1835ebd319afad0c031c36d03f36/yolox/utils/demo_utils.py
-import os
-import tempfile
-from typing import Optional
-from PIL import Image
 
+from PIL import Image
 import cv2
+from huggingface_hub import hf_hub_download
+from layoutparser.elements.layout_elements import TextBlock, Rectangle
+from layoutparser.elements.layout import Layout
 import numpy as np
 import onnxruntime
-from pdf2image import convert_from_path
 
-from unstructured_inference.inference.layout import DocumentLayout, LayoutElement, PageLayout
+from unstructured_inference.models.unstructuredmodel import UnstructuredModel
 from unstructured_inference.visualize import draw_bounding_boxes
-from unstructured_inference.models.base import UnknownModelException
-import json
-from typing import Tuple, Dict
-from huggingface_hub import hf_hub_download
+from unstructured_inference.utils import LazyDict, LazyEvaluateInfo
+
+YOLOX_LABEL_MAP = {
+    0: "Caption",
+    1: "Footnote",
+    2: "Formula",
+    3: "List-item",
+    4: "Page-footer",
+    5: "Page-header",
+    6: "Picture",
+    7: "Section-header",
+    8: "Table",
+    9: "Text",
+    10: "Title",
+}
+
+MODEL_TYPES = {
+    "yolox": LazyDict(
+        model_path=LazyEvaluateInfo(
+            hf_hub_download, "unstructuredio/yolo_x_layout", "yolox_l0.05.onnx"
+        ),
+        label_map=YOLOX_LABEL_MAP,
+    ),
+    "yolox_tiny": LazyDict(
+        model_path=LazyEvaluateInfo(
+            hf_hub_download, "unstructuredio/yolo_x_layout", "yolox_tiny.onnx"
+        ),
+        label_map=YOLOX_LABEL_MAP,
+    ),
+}
+
+
+class UnstructuredYoloXModel(UnstructuredModel):
+    def predict(self, x: Image):
+        super().predict(x)
+        return self.image_processing(x)
+
+    def initialize(self, model_path: str, label_map: dict):
+        self.model = model_path
+        self.layout_classes = label_map
+
+    def image_processing(
```

Comment:
The deleted image_processing allows reusing the onnxruntime.InferenceSession (receives it as an argument), would be great to recover that for avoiding reloading the model.

Reply:
Ah, thanks for flagging this.

Reply:
I made some changes in [620550f](https://github.com/Unstructured-IO/unstructured-inference/pull/56/commits/620550f0171349e7ab21ee17f6e59c90cc334c01) to persist the inference session in the model object. See if that addresses your concern here.

Reply:
Yes, I was thinking something like that.